src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package mate.academy.springbootwebgreqit.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootwebgreqit.dto.cartItem.CartItemDto;
+import mate.academy.springbootwebgreqit.dto.shoppingCart.ShoppingCartDto;
+import mate.academy.springbootwebgreqit.exception.EntityNotFoundException;
+import mate.academy.springbootwebgreqit.mapper.CartItemMapper;
+import mate.academy.springbootwebgreqit.mapper.ShoppingCartMapper;
+import mate.academy.springbootwebgreqit.model.CartItem;
+import mate.academy.springbootwebgreqit.model.ShoppingCart;
+import mate.academy.springbootwebgreqit.model.User;
+import mate.academy.springbootwebgreqit.repository.CartItemRepository;
+import mate.academy.springbootwebgreqit.repository.ShoppingCartRepository;
+import mate.academy.springbootwebgreqit.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getShoppingCartForCurrentUser(User user) {
+        ShoppingCart shoppingCart = user.getShoppingCart();
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addBookToShoppingCart(CartItemDto cartItemDto, User user) {
+        ShoppingCart shoppingCart = user.getShoppingCart();
+
+        CartItem existingItem = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().equals(cartItemDto.getBook()))
+                .findFirst()
+                .orElse(null);
+
+        if (existingItem != null) {
+            existingItem.setQuantity(existingItem.getQuantity() + cartItemDto.getQuantity());
+        } else {
+            CartItem newCartItem = cartItemMapper.toModel(cartItemDto);
+            newCartItem.setShoppingCart(shoppingCart);
+            shoppingCart.getCartItems().add(newCartItem);
+        }
+
+        shoppingCartRepository.save(shoppingCart);
+
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+
+    @Override
+    public ShoppingCartDto updateCartItemQuantity(Long cartItemId, int quantity, User user) {
+        ShoppingCart shoppingCart = user.getShoppingCart();
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't update cart item by quantity"));
+        cartItem.setQuantity(quantity);
+        cartItemRepository.save(cartItem);
+
+         shoppingCart = cartItem.getShoppingCart();
```

Comment:
```suggestion
        shoppingCart = cartItem.getShoppingCart();
```
please use Ctrl Alt L on shole src/main